pandas/core/apply.py

```diff
@@ -284,6 +283,13 @@ def transform_str_or_callable(self, func) -> DataFrame | Series:
         except Exception:
             return func(obj, *args, **kwargs)
 
+    def _filter_numeric_only(self) -> list[Any]:
+        if "numeric_only" in self.kwargs and bool(self.kwargs["numeric_only"]) is True:
```

Comment:
Instead of 

    and bool(self.kwargs["numeric_only"]) is True 

why not just 

    and self.kwargs["numeric_only"]